tests/test_runtime_http.cpp

```
@@ -1446,6 +1446,71 @@ TEST(HttpSSRF, PublicAddrInfoSynthetic) {
+
+TEST(HttpSSRF, PublicHostAboveCarrierGradeNAT) {
+    EXPECT_FALSE(__ry_is_private_host("100.128.0.0", 80));
+}
+
+TEST(HttpSSRF, PublicHostBelowMulticast) {
+    EXPECT_FALSE(__ry_is_private_host("223.255.255.255", 80));
+}
+
+// --- Additional SSRF range tests (IPv6 synthetic) ---
+
+TEST(HttpSSRF, PrivateAddrIPv6Unspecified) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)